### PR TITLE
Fix paths in CODEOWNERS

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
This PR fixes paths in CODEOWNERS that were violating following two rules, given [here](https://github.com/Azure/azure-sdk/blob/main/docs/policies/opensource.md#codeowners):

1. All paths must start with `/`
2. Paths to directories must end with `/`

These changes should have no effect on [owners determination by GitHub](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS).

Violations of rule 2. would result in [ADO build failure notifications](https://github.com/Azure/azure-sdk/blob/main/docs/policies/opensource.md#codeowners) being set to different set of people once we enable the new regex-based CODEOWNERS matcher, per:
- https://github.com/Azure/azure-sdk-tools/pull/5165

This PR is done in preparation of following work:

- https://github.com/Azure/azure-sdk-tools/issues/4859
- https://github.com/Azure/azure-sdk-tools/pull/5088
- https://github.com/Azure/azure-sdk-tools/issues/2770

This PR is analogous to its equivalent for `azure-sdk-for-net` repo:
- https://github.com/Azure/azure-sdk-for-net/pull/33584

I obtained the data required to make this PR by running tools added in this PR:
- https://github.com/Azure/azure-sdk-tools/pull/5245

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -43,8 +53,7 @@
 /sdk/appconfiguration/                                               @xiangyan99 @YalinLi0312
 
 # PRLabel: %App Configuration Provider
-/sdk/appconfiguration/azure-appconfiguration-provider/
-@mametcal @albertofori @avanigupta @mrm9084
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:47]: @jsquire @mrm9084 this path was completely broken.

Comment (mrm9084) [reply .github/CODEOWNERS:47]: Was? I am unable to see a difference between this and the change.

Comment (mrm9084) [reply .github/CODEOWNERS:47]: Also the path looks correct.

Comment (jsquire) [reply .github/CODEOWNERS:47]: Other than the newline at the end of what was 46, these look identical to me.

**Review by jsquire — APPROVED** [review]

File: .github/CODEOWNERS (hunk 1 above)
Comment (weshaggard) [reply .github/CODEOWNERS:47]: That is what was broken you cannot split the owners and the paths on separate lines.

**Pull request merged**